packages/toolbox-core/src/toolbox_core/sync_client.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+from threading import Thread
+from typing import Any, Awaitable, Callable, Mapping, Optional, TypeVar, Union
+
+from aiohttp import ClientSession
+
+from .client import ToolboxClient
+from .sync_tool import ToolboxSyncTool
+
+T = TypeVar("T")
+
+
+class ToolboxSyncClient:
+    """
+    A synchronous client for interacting with a Toolbox service.
+
+    Provides methods to discover and load tools defined by a remote Toolbox
+    service endpoint, returning synchronous tool wrappers (`SyncToolboxTool`).
+    It manages an underlying asynchronous `ToolboxClient`.
+    """
+
+    __session: Optional[ClientSession] = None
+    __loop: Optional[asyncio.AbstractEventLoop] = None
+    __thread: Optional[Thread] = None
+
+    def __init__(
+        self,
+        url: str,
+    ):
+        """
+        Initializes the SyncToolboxClient.
+
+        Args:
+            url: The base URL for the Toolbox service API (e.g., "http://localhost:8000").
```

Comment:
The default port for Toolbox service.

```suggestion
            url: The base URL for the Toolbox service API (e.g., "http://localhost:5000").
```